web/packages/teleport/src/Discover/Kubernetes/HelmChart/HelmChart.tsx

```diff
@@ -303,19 +303,31 @@ const generateCmd = (data: {
   clusterVersion: string;
   resourceId: string;
   isEnterprise: boolean;
+  isCloud: boolean;
+  automaticUpgradesEnabled: boolean;
 }) => {
+  let extraYAMLConfig = '';
+
+  if (data.isEnterprise) {
+    extraYAMLConfig += 'enterprise: true\n';
+  }
+
+  if (data.isCloud && data.automaticUpgradesEnabled) {
+    extraYAMLConfig += 'updater:\n';
+    extraYAMLConfig += '    enabled: true\n';
+    extraYAMLConfig += '    releaseChannel: "cloud/stable"\n';
+  }
+
   return `cat << EOF > prod-cluster-values.yaml
 roles: kube
 authToken: ${data.tokenId}
 proxyAddr: ${data.proxyAddr}
 kubeClusterName: ${data.clusterName}
-teleportVersionOverride: ${data.clusterVersion}
-enterprise: ${data.isEnterprise}
 labels:
     teleport.internal/resource-id: ${data.resourceId}
-EOF
+${extraYAMLConfig}EOF
```

Comment:
can we enable replication and pod distruption budget for everyone? we would need it at least if `data.automaticUpgradesEnabled`. Would you prefer to do it in another PR?

Reply:
I think we should do it in another PR.
Just because we should try to expose as less params as possible in those scripts to improve UX.

As for the change itself, if they depend then we should ensure that at the helm chart level, right?
There's also an option to set those values as default.

I can open a PR for that, but I'll need your review

Reply:
I don't think we want to enable HA by default on the chart, this would be a significant breaking change and according to @webvictim a non-negligible amount of users don't want nor need replication.

The chart could have a smart default system like "default to 2 when the updater is on, default to 1 if it is not, and follow the user input whatever happens," but it makes things less clear from a user perspective and is also absolutely terrible to implement in Helm.

We do need to advise cloud users to enable replication if they are enrolled in auto upgrades. I don't know about cloud users without auto upgrades.

For those reasons, I would address the HA in Teleport Discover for cloud users.

Reply:
I'll add HA like you suggested on the linked issue
```
highAvailability:
  replicaCount: 2
  podDisruptionBudget:
    enabled: true
    minAvailable: 1
```

> I don't know about cloud users without auto upgrades.

For auto upgrades to be on, it'll need to be a cloud cluster **and** the env var `TELEPORT_AUTOMATIC_UPGRADES` is set to a truthy value